keylime/tenant.py

```diff
@@ -1462,20 +1457,6 @@ def main() -> None:
         default=None,
         help="Specify the SHA-256 checksum of a runtime policy",
     )
-    parser.add_argument(
-        "--runtime-policy-sig",
-        action="store",
-        dest="runtime_policy_sig",
-        default=None,
-        help="Specify the signature file of a runtime policy",
-    )
-    parser.add_argument(
-        "--runtime-policy-sig-url",
-        action="store",
-        dest="runtime_policy_sig_url",
-        default=None,
-        help="Specify the URL of the remote signature file of a runtime policy",
-    )
     parser.add_argument(
         "--runtime-policy-sig-key",
```

Comment:
I think we should remove the signature verification key from the options and move it into the tenant configuration file. It's kind of odd to pass a policy as a parameter and voluntarily pass a key along with it. If the key goes into the configuration file it would become mandatory that all policies are signed IF a key is provided. If no key is provided we could just extract the payload and proceed (without signature verification). Also, currently one could just extract the payload from the json and convert it to a runtime policy and pass it in without key and so bypass the whole signature verification. Once it's in the tenant config file it could at least become mandatory that one one works with signed policies.

Reply:
If I understood it correctly, in an environment with diversity of nodes, we would end up with multiple configuration files just to deal with the "key" vs "no key" policies (and there is a variety of policies), correct? 
If yes, then I can safely say that, from an operations standpoint, I see this makes maintenance harder, not easier. We have been making a concerted effort to be able to work with configuration files with as little configuration as possible, with all customizations coming from either CLI options or environments variables.

Reply:
@stefanberger That's not correct - there **IS** a current configuration option to require signatures on runtime policies. The only difference from your proposal is that it's a verifier-side config option. The specific option is [`require_allow_list_signatures`](https://github.com/keylime/keylime/blob/master/keylime.conf#L383); it currently defaults to `false`, but if set, it'll check for a runtime policy's signature as it arrives on the verifier, and will return a `401` error code if the signature doesn't pass. This PR makes use of that option as well, and will check for DSSE signatures.

In any case, the current tenant-side logic in this pull request is to always verify the signature if it's detected as a DSSE envelope. The only way to not perform a signature check is to pass an unsigned policy.

Beyond that, another reason I'd err towards the CLI option is that the presence of a public key is not necessarily indicative of a signed policy. As mentioned in the description, it's possible to embed x509 certificates with an in-band public key into Keylime DSSE policies, which means you wouldn't need to pass a separate public key to verify the signature. In that case, you'd want to verify the envelope regardless, even without a path to a public key on disk.

Reply:
Ah. It doesn't seem to make it into the verifier.conf file, though.